Fix null handling in filters regression

sbelak opened the pull request:
Fixes a regression where the code emitted for `:does-not-contain` & co didn't correctly account for fields wrapped in `joined-fields`.

Also fixes Cypress test for the original feature.

Fixes #13534

sbelak committed:
Unskip cypress

nemanjaglumac commented:
@sbelak The right test to unskip for this particular issue is: https://github.com/metabase/metabase/commit/695ad0ef21057397308856f598a5a775a1f13f8a

sbelak committed:
Make null behaviour correction work with `joined-field`s

sbelak committed:
Fix cypress test

sbelak committed:
Make linter happy

sbelak commented:
Cypress fail seems to be some inconsistency with when we pluralize table names. On Nemanja and my machine the joined table is named Product, while on CI it's Products.

sbelak committed:
Cypress fix

camsaul commented:
I have seen some Cypress tests that check against a regex (e.g. `/products?/i`) and mention in adjacent comments the inconsistent pluralization is the reason why. I think @flamber added that stuff or knows more?

flamber commented:
Yeah, I've been seeing a random mix of singular/plural on all my different setups. Haven't figured out the reason why yet, but I guess it relates to the Admin > Settings > General > Friendly Table and Field Names
Anyways, regex is definitely the way to go.

nemanjaglumac commented:
> I have seen some Cypress tests that check against a regex (e.g. `/products?/i`) and mention in adjacent comments the inconsistent pluralization is the reason why. I think @flamber added that stuff or knows more?

Yup. I made one such test after @flamber  warned me that he is not sure how the name will be "normalized". I understood then that it was for that specific case only. Didn't know that this is a global "issue", especially since I didn't run into this problem so far.

So, if you guys think that regex is the way to go, I can continue writing tests like that?

nemanjaglumac committed:
Fix Cypress test

- added regex pattern for weird singular/plural issue
- added a different positive assertion at the end

nemanjaglumac commented:
Unrelated Cypress test flaked. I hit the "rerun from failed".
